File: Practica1/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author Lorena
 */
public class Casilla {
    private int numeroCasilla; //indica numero de casilla
    private int coste; //indica el coste de compra del titulo de la propiedad de esa casilla
    private int numHoteles, numCasas; // inidca el numero de hoteles y de cassas edificados en esa casilla
    private TipoCasilla tipo; // indica el tipo de la casilla
    private TituloPropiedad titulo; //para asociar la casilla a su titulo de propiedad
    
    //Constructores:
    //Para los que son de tipo calle
    Casilla(int numeroCasilla, int coste, TipoCasilla tipo, TituloPropiedad titulo){
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        numHoteles = 0;
        numCasas = 0;
        this.tipo = tipo;
        this.titulo = titulo;
    }
    
    //Para los que NO son de tipo calle
    Casilla(int numeroCasilla, TipoCasilla tipo){
        this.numeroCasilla = numeroCasilla;
        this.tipo = tipo;
    }
    
    //Modificadores
    void setNumHoteles(int numHoteles){
        this.numHoteles = numHoteles;
    }
    
    void setNumCasas(int numCasas){
        this.numCasas = numCasas;
    }
    
    //Consultores
    int getNumeroCasilla(){
        return numeroCasilla;
    }
    
    int getCoste(){
        return coste;
    }
    
    int getNumHoteles(){
        return numHoteles;
    }
    
    int getNumCasas(){
        return numCasas;
    }
    
    TipoCasilla getTipo(){
        return tipo;
    }
    
    TituloPropiedad getTitulo(){
        return titulo;
    }
    
    //Método toString() devuelve un String con el estado del objeto correspondiente
    @Override
    public String toString(){
        if(titulo != null){
            return "\n Numero de la Casilla =" + Integer.toString(numeroCasilla)
                    + "\n Coste = " + Integer.toString(coste)
                    + "\n Numero de hoteles =" + Integer.toString(numHoteles) 
                    + "\n Numero de casas =" + Integer.toString(numCasas)
                    + "\n Tipo de la casilla =" + tipo 
                    + "\n Titulo de propiedad =" + titulo.toString()
                    + "\n---------------------------------------------------------";
        }
        else{
            return "\n Numero de la Casilla =" + Integer.toString(numeroCasilla)
                + "\n Tipo de la casilla =" + tipo
                + "\n---------------------------------------------------------";

        }
    }

}

File: Practica1/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'tipo_casilla'
require_relative 'titulo_propiedad'

module ModeloQytetet
  class Casilla
    #Consultores y modificadores
    attr_accessor :numHoteles, :numCasas, :titulo
    attr_reader :numeroCasillas, :coste, :tipo
    
    #Constructores
    #Para los que son de tipo calle
    def initialize(numeroCasilla, coste, tipo, titulo)
      @numeroCasilla = numeroCasilla
      @coste = coste
      @tipo = tipo
      @titulo = titulo
      @numHoteles = 0
      @numCasas = 0
    end
    #Para los que NO son de tipo calle
    def self.noCalle(numeroCasilla, tipo)
      Casilla.new(numeroCasilla, 0, tipo, 0)
    end
    
    #Método toString devuelve un String con el estado del objeto correspondiente
    def to_s
      if(@coste !=0)
        "\n Numero de la casilla: #{@numeroCasilla} 
        \n Coste: #{@coste} 
        \n Numero de hoteles: #{@numHoteles}
        \n Numero de casas: #{@numCasas}
        \n Tipo de la casilla: #{@tipo}
        \n Titulo de propiedad: #{@titulo}
        \n-----------------------------------------\n"
      else
      "\n Numero de la casilla: #{@numeroCasilla}
      \n Tipo de la casilla: #{@tipo}
      \n-----------------------------------------\n"
      end
    end 
    private :titulo=
  end
end
